server/src/main/java/org/elasticsearch/rest/action/search/CCSVersionCheckHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.rest.action.search;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.VersionCheckingStreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.Collections;
+import java.util.List;
+
+public class CCSVersionCheckHelper {
+
+    public static String CCS_VERSION_CHECK_FLAG = "ccs_force_fail";
```

Comment:
I would propose that we make this a naming discussion :) 

Shall we call it `check_ccs_compatibility` ? I am tempted to make it accept a version...but maybe that complicates things for consumer that end up having to make a decision. And I don't love the idea of accepting both true and a version.